exercises/binary-search/src/test/java/BinarySearchTest.java

```diff
@@ -0,0 +1,100 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class BinarySearchTest {
+
+    private static final List<Integer> SORTED_LIST
+            = Collections.unmodifiableList(
+                    Arrays.asList(1, 2, 3, 4, 5, 6)
+            );
+
+    public static final List<Integer> SORTED_LIST_OF_ODD_LENGTH
+            = Collections.unmodifiableList(
+                    Arrays.asList(0, 1, 2, 2, 3, 10, 12)
+            );
+
+    public static final List<Integer> UNSORTED_LIST
+            = Collections.unmodifiableList(
+                    Arrays.asList(10, 2, 5, 1)
+            );
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
```

Comment:
Removed the @Rule from the tests altogether as according to the [canonical test data](https://github.com/exercism/x-common/blob/master/exercises/binary-search/canonical-data.json) one shouldn't test if the array is sorted.